src/apint/arithmetic.rs

```diff
@@ -53,154 +77,184 @@ impl ApInt {
+    pub fn into_wrapping_add(self, rhs: &ApInt) -> Result<ApInt> {
+        try_forward_bin_mut_impl(self, rhs, ApInt::wrapping_add_assign)
+    }
+
+    /// Add-assigns `rhs` to `self` inplace, and returns a boolean indicating if overflow occured,
+    /// according to the **unsigned** interpretation of overflow.
     /// 
     /// # Errors
     /// 
     /// - If `self` and `rhs` have unmatching bit widths.
-    pub fn into_wrapping_sub(self, rhs: &ApInt) -> Result<ApInt> {
-        try_forward_bin_mut_impl(self, rhs, ApInt::wrapping_sub_assign)
+    pub fn overflowing_uadd_assign(&mut self, rhs: &ApInt) -> Result<bool> {
+        match self.width().excess_bits() {
+            Some(excess) => {
+                let mask = Digit::all_set() >> excess;
+                match self.zip_access_data_mut_self(rhs)? {
+                    Inl(lhs, rhs) => {
+                        let temp = lhs.wrapping_add(rhs);
+                        *lhs = temp & mask;
+                        //excess bits are cleared by the mask
+                        return Ok((temp & mask) != temp)
+                    }
+                    Ext(lhs, rhs) => {
+                        let (temp, mut carry) = lhs[0].carrying_add(rhs[0]);
+                        lhs[0] = temp;
+                        for i in 1..(lhs.len() - 1) {
+                            let temp = lhs[i].dd().wrapping_add(rhs[i].dd()).wrapping_add(carry.dd());
+                            lhs[i] = temp.lo();
+                            carry = temp.hi();
+                        }
+                        let temp = lhs[lhs.len() - 1]
+                            .wrapping_add(rhs[lhs.len() - 1])
+                            .wrapping_add(carry);
+                        lhs[lhs.len() - 1] = temp & mask;
+                        //excess bits are cleared by the mask
+                        return Ok((temp & mask) != temp)
+                    }
+                }
+            }
+            None => {
+                match self.zip_access_data_mut_self(rhs)? {
+                    Inl(lhs, rhs) => {
+                        let temp = lhs.overflowing_add(rhs);
+                        *lhs = temp.0;
+                        //no excess bits to clear
+                        return Ok(temp.1)
+                    }
+                    Ext(lhs, rhs) => {
+                        let (temp, mut carry) = lhs[0].carrying_add(rhs[0]);
+                        lhs[0] = temp;
+                        for i in 1..lhs.len() {
+                            let temp = lhs[i].dd().wrapping_add(rhs[i].dd()).wrapping_add(carry.dd());
+                            lhs[i] = temp.lo();
+                            carry = temp.hi();
+                        }
+                        //no excess bits to clear
+                        return Ok(carry != Digit::zero())
+                    }
+                }
+            }
+        }
     }
 
+    //pub(crate) fn overflowing_sadd_assign()
```

Comment:
There is a difference between signed and unsigned overflow however. e.x. if we have two four bit `ApInt`s, one of which is 1110 (-2 in the signed interpretation) and another 1110, and we add them, it will make 1100 (-4). There are carries off the end but not numerical overflow

Reply:
`ApInt` as I have designed it is unrelated to signedness. So only operations defined on it are either signed or unsigned operations. So there is basically no difference between `uadd` and `sadd` in the general case and if we have the overflowing version that tells us if there is an overflow bit we do not need one without that in my opinion. I hope I understood this correctly.

Reply:
Note that `Int` and `Uint` are existent to provide a signed and unsigned version and API for the underlying `ApInt` exactly for the reason stated above.

Reply:
I still think there should be two functions, because testing for signed overflow would involve
```
let overflow = lhs.overflowing_add_assign(rhs).unwrap();
signed_overflow = (lhs.is_negative() == rhs.is_negative()) && overflow;
```
which I would put in a overflowing_sadd_assign